application/controllers/ApiV1ContactsController.php

```diff
@@ -0,0 +1,604 @@
+                            }
+
+                            $condition->setColumn('external_uuid');
+                        }
+                    }
+                )->parse()
+        );
+
+        switch ($method) {
+            case 'GET':
+                $stmt = (new Select())
+                    ->distinct()
+                    ->from('contact co')
+                    ->columns([
+                        'contact_id'        => 'co.id',
+                        'id'                => 'co.external_uuid',
+                        'full_name',
+                        'username',
+                        'default_channel'   => 'ch.name',
+                    ])
+                    ->joinLeft('contact_address ca', 'ca.contact_id = co.id')
+                    ->joinLeft('channel ch', 'ch.id = co.default_channel_id');
+
+                if ($identifier !== null) {
+                    $stmt->where(['external_uuid = ?' => $identifier]);
+
+                    /** @var stdClass|false $result */
+                    $result = $db->fetchOne($stmt);
+
+                    if ($result === false) {
+                        $this->httpNotFound('Contact not found');
+                    }
+
+                    if ($result->username === null) {
+                        unset($result->username);
+                    }
+
+                    $groups = $this->fetchGroupIdentifiers($result->contact_id);
+                    if ($groups) {
+                        $result->groups = $groups;
+                    }
+
+                    $addresses = $this->fetchContactAddresses($result->contact_id);
+                    if ($addresses) {
+                        $result->addresses = $addresses;
+                    }
+
+                    unset($result->contact_id);
+                    $results[] = $result;
+
+                    break;
+                }
+
+                if ($filter !== null) {
+                    $stmt->where($filter);
+                }
+
+                $stmt->limit(500);
+                $offset = 0;
+
+                ob_end_clean();
+                Environment::raiseExecutionTime();
+
+                $this->getResponse()
+                    ->setHeader('Content-Type', 'application/json')
+                    ->setHeader('Cache-Control', 'no-store')
+                    ->sendResponse();
+
+                echo '[';
+
+                $res = $db->select($stmt->offset($offset));
+                do {
+                    /** @var stdClass $row */
+                    foreach ($res as $i => $row) {
+                        if ($row->username === null) {
+                            unset($row->username);
+                        }
+
+                        $groups = $this->fetchGroupIdentifiers($row->contact_id);
+                        if ($groups) {
+                            $row->groups = $groups;
+                        }
```

Comment:
Same as for the groups above. Do not omit the property but return it with its empty state.